receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/accumulator.go

```diff
@@ -0,0 +1,105 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awsecscontainermetrics
+
+import (
+	"time"
+
+	metricspb "github.com/census-instrumentation/opencensus-proto/gen-go/metrics/v1"
+	resourcepb "github.com/census-instrumentation/opencensus-proto/gen-go/resource/v1"
+	"github.com/golang/protobuf/ptypes/timestamp"
+	"go.opentelemetry.io/collector/consumer/consumerdata"
+)
+
+// metricDataAccumulator defines the accumulator
+type metricDataAccumulator struct {
+	md []*consumerdata.MetricsData
+}
+
+// getMetricsData generates OT Metrics data from task metadata and docker stats
+func (acc *metricDataAccumulator) getMetricsData(containerStatsMap map[string]ContainerStats, metadata TaskMetadata) {
+
+	var taskMemLimit uint64
+	var taskCPULimit float64
+	taskMetrics := ECSMetrics{}
+	timestamp := timestampProto(time.Now())
+	taskLabelKeys, taskLabelValues := taskLabelKeysAndValues(metadata)
+
+	for _, containerMetadata := range metadata.Containers {
+		stats := containerStatsMap[containerMetadata.DockerID]
+		containerMetrics := getContainerMetrics(stats)
+		containerMetrics.MemoryReserved = *containerMetadata.Limits.Memory
+		containerMetrics.CPUReserved = *containerMetadata.Limits.CPU
+
+		taskMemLimit += containerMetrics.MemoryReserved
+		taskCPULimit += containerMetrics.CPUReserved
+
+		labelKeys, labelValues := containerLabelKeysAndValues(containerMetadata)
+		labelKeys = append(labelKeys, taskLabelKeys...)
+		labelValues = append(labelValues, taskLabelValues...)
+
+		acc.accumulate(
+			timestamp,
+			convertToOTMetrics(ContainerPrefix, containerMetrics, labelKeys, labelValues, timestamp),
+		)
+
+		aggregateTaskMetrics(&taskMetrics, containerMetrics)
+	}
+
+	// Overwrite Memory limit with task level limit
+	if metadata.Limits.Memory != nil {
+		taskMetrics.MemoryReserved = *metadata.Limits.Memory
+	}
+
+	taskMetrics.CPUReserved = taskMetrics.CPUReserved / CPUsInVCpu
+
+	// Overwrite CPU limit with task level limit
+	if metadata.Limits.CPU != nil {
+		taskMetrics.CPUReserved = *metadata.Limits.CPU
+	}
+
+	acc.accumulate(
+		timestamp,
+		convertToOTMetrics(TaskPrefix, taskMetrics, taskLabelKeys, taskLabelValues, timestamp),
+	)
+}
+
+func (acc *metricDataAccumulator) accumulate(
+	startTime *timestamp.Timestamp,
+	m ...[]*metricspb.Metric,
+) {
+	var resourceMetrics []*metricspb.Metric
+	for _, metrics := range m {
+		for _, metric := range metrics {
+			if metric != nil {
+				metric.Timeseries[0].StartTimestamp = startTime
+				resourceMetrics = append(resourceMetrics, metric)
+			}
+		}
+	}
+
+	resourceAttributes := map[string]string{}
+	resourceAttributes[ResourceAttributeServiceNameKey] = ResourceAttributeServiceNameValue
+
+	r := &resourcepb.Resource{
+		Type:   MetricResourceType,
```

Comment:
As @anuraaga mentioned, this will be dropped eventually. However, if it does get utilized, we prefer to use `aoc.ecs` for all metrics we are receiving from this receiver.